dusty/systems/docker/compose.py

```diff
@@ -61,15 +73,21 @@ def _restart_container(client, container):
         log_to_client('Restarting {}'.format(_get_canonical_container_name(container)))
         client.restart(container['Id'], timeout=1)
 
+    assembled_specs = get_assembled_specs()
+    if services == []:
+        services = [spec.name for spec in assembled_specs.get_apps_and_services()]
     logging.info('Restarting service containers from list: {}'.format(services))
     client = get_docker_client()
-    dusty_containers = _get_dusty_containers(client, services)
-    expected_number_of_containers = get_expected_number_of_running_containers() if len(services) == 0 else len(services)
-    if len(dusty_containers) != expected_number_of_containers:
-        log_to_client("Not going to restart containers. Expected number of containers {} does not match {}".format(expected_number_of_containers, len(dusty_containers)))
-        raise RuntimeError("Please use `docker ps -a` to view crashed containers")
-    else:
-        for container in dusty_containers:
+    for service in services:
+        container = _get_container_for_app_or_service(client, service, include_exited=True)
+        if container is None:
+            log_to_client('No container found for {}'.format(service))
+            continue
+        stopped_linked_containers = _check_stopped_linked_containers(client, container, assembled_specs)
+        if stopped_linked_containers:
+            log_to_client('No running containers {0}, which are linked to by {1}.  Cannot restart {1}'.format(
+                              stopped_linked_containers, service))
```

Comment:
so what if a container has no links? won't this now be able to restart that container?